crates/storage-query-datafusion/src/log/row.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::table_util::format_using;
+
+use super::schema::{LogBuilder, LogRowBuilder};
+use restate_types::{
+    logs::{
+        LogId,
+        metadata::{ProviderKind, Segment},
+    },
+    replicated_loglet::ReplicatedLogletParams,
+};
+use tracing::debug;
+
+#[inline]
+pub(crate) fn append_segment_row(
+    builder: &mut LogBuilder,
+    output: &mut String,
+    id: LogId,
+    segment: &Segment<'_>,
+) {
+    if segment.config.kind == ProviderKind::Replicated {
+        match ReplicatedLogletParams::deserialize_from(segment.config.params.as_bytes()) {
```

Comment:
Can you lookup the index instead of deserializing? check `Logs::get_replicated_loglet()`